plugins/devel/event-testing.cpp

```diff
@@ -0,0 +1,164 @@
+#include "Core.h"
```

Comment:
I meant more a `devel/eventful.lua` script that just registers handlers for everything and echoes to the screen when the callbacks are called. This would at least allow you to manually verify that the event detection -> event triggering -> Lua marshalling chain basically functions for all event types. I would expect that kind of test to have been run before the code is merged. Some of the types we're working with here are not well understood, and we're making a significant (and risky) change to the API in that we're allowing events to fire in a tick other than the one where the event was detected.

Writing automated regression tests that trigger all the events would be very challenging, but it might be possible for a few event types. It could be done for building creation pretty easily, for example. Quickfort even has an API that you can use to create and destroy buildings with a single call.

Reply:
ah, ok a devel/ script definitely doesn't seem impossible. I'll try to write one before the end of the month, since it's lua I'm less keen